### PR TITLE
HBASE-28704 The expired snapshot can be read by CopyTable or ExportSnapshot

### PR DESCRIPTION
**Pull request by guluo2016**
Details see: [HBASE-28704](https://issues.apache.org/jira/browse/HBASE-28704)

### CODE REVIEW

File: hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshot.java [hunk 1]
```diff
@@ -281,6 +283,39 @@ public void testExportWithResetTtl() throws Exception {
     }
   }
 
+  @Test
```
Comment (guluo2016) [inline hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshot.java:286]: By default, this test case does not execute  beacuse of  https://github.com/apache/hbase/blob/master/hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/snapshot/TestExportSnapshot.java#L78

After manually modifying it, the test case passes.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 27s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 56s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m 32s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 45s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 57s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 28s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 28s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 45s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m  9s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |   5m 11s |  |  Patch does not cause any errors with Hadoop 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 40s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 17s |  |  The patch does not generate ASF License warnings.  |
|  |   |  32m 34s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6047 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 721d4a07ba6a 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 412dc8489d2f9104da0407cb7bd7d25c5fa5a582 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 83 (vs. ulimit of 30000) |
| modules | C: hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m  7s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 15s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 40s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 42s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 57s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 14s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 14s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 44s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 209m  9s |  |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  17m 45s |  |  hbase-mapreduce in the patch passed.  |
|  |   | 253m 22s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6047 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux 9a5590952a63 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 412dc8489d2f9104da0407cb7bd7d25c5fa5a582 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/1/testReport/ |
| Max. process+thread count | 5194 (vs. ulimit of 30000) |
| modules | C: hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]
@2005hithlj Please also take a look?

Thanks.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 26s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 52s |  |  master passed  |
| +1 :green_heart: |  compile  |   3m 29s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 47s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   1m 53s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 26s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 26s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 45s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m  7s |  |  the patch passed  |
| -1 :x: |  hadoopcheck  |   5m 53s |  |  The patch causes 10 errors with Hadoop v3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 42s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 17s |  |  The patch does not generate ASF License warnings.  |
|  |   |  28m  9s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6047 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux f0f41a502c88 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 19132baa5038e7821c8926cfaf244cb891bce239 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| hadoopcheck | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/2/artifact/yetus-general-check/output/patch-javac-3.4.0.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/2/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 26s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 35s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 40s |  |  master passed  |
| +1 :green_heart: |  compile  |   1m 14s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 42s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 53s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 57s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 14s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 14s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 39s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 42s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 226m  8s |  |  hbase-server in the patch passed.  |
| +1 :green_heart: |  unit  |  21m 11s |  |  hbase-mapreduce in the patch passed.  |
|  |   | 275m 58s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/2/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6047 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux da3332192074 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 19132baa5038e7821c8926cfaf244cb891bce239 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/2/testReport/ |
| Max. process+thread count | 5439 (vs. ulimit of 30000) |
| modules | C: hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/2/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  1s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 55s |  |  master passed  |
| +1 :green_heart: |  compile  |   4m 49s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 58s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m 48s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 56s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 40s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 27s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 27s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m 11s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m 56s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  11m 45s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 59s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 22s |  |  The patch does not generate ASF License warnings.  |
|  |   |  48m 21s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6047 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux e99b07ba8aa8 5.4.0-186-generic #206-Ubuntu SMP Fri Apr 26 12:31:10 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 19132baa5038e7821c8926cfaf244cb891bce239 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 84 (vs. ulimit of 30000) |
| modules | C: hbase-server hbase-mapreduce U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6047/3/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
@2005hithlj Do you have any time to take a look at this?

Thanks.

**Review by 2005hithlj — APPROVED** [review]
LGTM +1

**Pull request merged**